crates/sparrow-instructions/src/evaluators/aggregation/token/collect_struct_token.rs

```diff
@@ -0,0 +1,47 @@
+use arrow::array::{new_null_array, Array, ArrayRef, AsArray};
+
+use crate::{ComputeStore, StateToken, StoreKey};
+
+/// Token used for collecting structs
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+pub struct CollectStructToken {
+    /// Stores the state for in-memory usage.
+    ///
+    /// A [ListArray] comprised of lists of structs for each entity.
+    #[serde(with = "sparrow_arrow::serde::array_ref")]
+    pub state: ArrayRef,
```

Comment:
Eventulaly, we may be able to just have a single `ArrayRefStateToken` for any of these that use an Arrow array as state.

Reply:
Yes, agreed. Would love to spend some time to simplify the aggregations, tokens, and agg functions.